app/models/html_variant.rb

```diff
@@ -10,8 +10,8 @@ class HtmlVariant < ApplicationRecord
   validate  :no_edits
 
   belongs_to :user, optional: true
-  has_many :html_variant_trials
-  has_many :html_variant_successes
+  has_many :html_variant_trials, dependent: :destroy
+  has_many :html_variant_successes, dependent: :destroy
```

Comment:
trials and successes are strongly coupled with the variants, so they should just go